future-apps/payroll/test/payroll_allowed_tokens.js

```diff
@@ -0,0 +1,95 @@
+const { assertRevert } = require('@aragon/test-helpers/assertThrow')
+
+const getContract = name => artifacts.require(name)
+const getEvent = (receipt, event, arg) => { return receipt.logs.filter(l => l.event == event)[0].args[arg] }
+
+contract('Payroll, allowed tokens,', function(accounts) {
+  const [owner, employee] = accounts
+  const {
+    deployErc20TokenAndDeposit,
+    addAllowedTokens,
+    redistributeEth,
+    getDaoFinanceVault,
+    initializePayroll
+  } = require("./helpers.js")(owner)
+
+  let payroll
+  let payrollBase
+  let priceFeed
+  let dao
+  let finance
+  let vault
+
+  let erc20Tokens = []
+  const ERC20_TOKEN_DECIMALS = 18
+  const SALARY = (new web3.BigNumber(1)).times(10 ** Math.max(ERC20_TOKEN_DECIMALS - 4, 1))
+  const RATE_EXPIRY_TIME = 1000
+  const NOW_MOCK = new Date().getTime()
+  let ETH
+  let MAX_ALLOWED_TOKENS
+
+  const MAX_GAS_USED = 7e6
```

Comment:
```suggestion
  const MAX_GAS_USED = 6e6
```

Reply:
Let's play it a bit safer, I don't like functions that need more than 80% of the current block gas limit